internal/api/plugin/plugin.go

```diff
@@ -17,18 +17,39 @@ import (
 	"encoding/json"
 	"errors"
 	"os"
-	"path"
+	"path/filepath"
 
 	"github.com/perses/perses/pkg/model/api/config"
 	"github.com/sirupsen/logrus"
 )
 
-const pluginFileName = "plugin.json"
+const (
+	PluginModuleKind = "PluginModule"
+	pluginFileName   = "plugin-modules.json"
+)
+
+type PluginMetadata struct {
```

Comment:
I propose to put that in `pkg/model/api/v1` and then to use the classic `Kind, Metadata, Spec`

Then in spec, you can have the Display struct and the list of the plugins

Reply:
I tried to use the default Metadata but the version there is numeric, and for the case of the plugins it will use semver. Should we change it there? Am I missing something?

Reply:
ah yeah I forgot it contains a version in number.

So yeah let's create another metadata for the plugin only. But I would still consider to use the classic struct `Kind`, `Metadata`, `Spec` even if `Metadata` is another one.

Something like that

```go
type PluginSpec {
  Display *common.Display
  Plugins []Plugin
}

type Plugin struct {
  Kind Kind
  Metadata PluginMetadata
  Spec PluginSpec
}
```